ql/cashflows/overnightindexedcoupon.cpp

```diff
@@ -83,25 +87,38 @@ namespace QuantLib {
                     }
                 }
 
+                auto endIndex = i;
+
+                if (minDate < valueDates[n]) {
+                    while (endIndex < n && minDate <= valueDates[endIndex])
+                        endIndex++;
+                } else {
+                    endIndex = n;
+                }
+
```

Comment:
I think this can be moved before the whole calculation, after which `endIndex` can be used instead of `n` even in the fixed part.  It should make it unnecessary to use `minDate`.  What do you think?

(Also, the loop can probably be replaced by one call to `std::lower_bound`)